api/src/main/java/org/ehcache/expiry/ExpiryPolicy.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.expiry;
+
+import org.ehcache.ValueSupplier;
+
+import java.time.Duration;
+
+/**
+ * A policy object that governs expiration for mappings in a {@link org.ehcache.Cache Cache}.
+ * <p>
+ * Previous values are not accessible directly but are rather available through a {@link ValueSupplier value supplier}
+ * to indicate that access can require computation (such as deserialization).
+ * <p>
+ * {@link Duration#isNegative() Negative duration} are not supported, expiry policy implementation returning such a
+ * duration will result in immediate expiry, as if the duration was {@link Duration#ZERO zero}.
+ * <p>
+ * NOTE: Some cache configurations (eg. caches with eventual consistency) may use local (ie. non-consistent) state
+ * to decide whether to call {@link #getExpiryForUpdate(Object, ValueSupplier, Object)}  vs.
+ * {@link #getExpiryForCreation(Object, Object)}. For these cache configurations it is advised to return the same
+ * value for both of these methods
+ * <p>
+ *
+ * @param <K> the key type for the cache
+ * @param <V> the value type for the cache
+ *
+ */
+public interface ExpiryPolicy<K, V> {
+
+  /**
+   * A {@link Duration duration} that represents an infinite time.
+   */
+  Duration INFINITE = Duration.ofSeconds(Long.MAX_VALUE);
```

Comment:
To be coherent with cluster timeouts (and be `duration.toNanos()` safe), this should be `Duration.ofNanos(Long.MAX_VALUE)`

Reply:
I understand the request ... not sure I like going that way though. I would feel better keeping at least `Duration.ofMillis(Long.MAX_VALUE)`.
@chrisdennis wdyt?

Reply:
I much prefer Duration.ofNanos to never crash on infinite. As mentioned, we might refactor that differently later. But right now, it's the safer side. You could probably even do what I did for timeouts and move expiration bigger than that to this value.

Reply:
So changing that is more work than expected ... as I hardcoded `INFINITE` to be `Duration.getSeconds() == Long.MAX_VALUE` in a number of places as a consequence of `or.ehcache.expiry.Duration.isInfinite()` going away. Fixing that mistake so that we have only a limited number of places where we "decide" what `INFINITE`.